File: Month00_CSharp/Workers/Incoming/ExecutionModel.cs
namespace Workers.Incoming;

internal enum ExecutionModel : byte
{
    OneOff      = 0, // Single fire-and-forget unit
    Scheduled   = 1, // Time-triggered (cron/delayed)
    EventDriven = 2, // Triggered by domain/infrastructure event
    Batch       = 3, // Large dataset processing
    Stream      = 4, // Long-lived continuous processing
    Workflow    = 5, // DAG / multistep orchestration
    Actor       = 6  // Key-partitioned stateful unit
}

File: 03_FSharp/Workers/Incoming/ExecutionModel.fs
namespace Workers.Incoming

type ExecutionModel =
    | OneOff = 0uy // Single fire-and-forget unit
    | Scheduled = 1uy // Time-triggered (cron/delayed)
    | EventDriven = 2uy // Triggered by domain/infrastructure event
    | Batch = 3uy // Large dataset processing
    | Stream = 4uy // Long-lived continuous processing
    | Workflow = 5uy // DAG / multistep orchestration
    | Actor = 6uy // Key-partitioned stateful unit
